### PR TITLE
Drag and drop and new ways of authentication

### PR DESCRIPTION
**Pull request by MaxDaniLOVE**
### Implemented:
- Add drag'n'drop;
- Add authentication using Github/Facebook/Google;
- Fix minor issues;

- [Demo](https://dims-5.netlify.app/)

### CODE REVIEW

File: src/components/ChangePassForm/changePassForm.scss [hunk 1]
```diff
@@ -20,7 +20,7 @@
 .dark-modal {
   #change-pass-modal {
     background-color: $charcoal;
-    .pass-hide-btn {
+    #pass-hide-btn {
```
Comment (maximsan) [inline src/components/ChangePassForm/changePassForm.scss:23]: Actually, it's better not to use id for styling without real necessity.

File: src/components/DraggableRow/DraggableRow.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+/* eslint-disable react/jsx-props-no-spreading */
+import React from 'react';
+import { Draggable } from 'react-beautiful-dnd';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+const DraggableRow = ({ children, draggableId, index, isDarkMode, isDragDisabled }) => {
+  const backgroundColor = isDarkMode ? '#2a2e36' : '#d4d3d3';
```
Comment (maximsan) [inline src/components/DraggableRow/DraggableRow.js:8]: it's more convenient to export colors as COLORS.<name>  from constants

File: src/components/DraggableTable/DraggableTable.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  };
+
+  const isDragDisabled = children.length <= 1;
+
+  const tableRows = React.Children.map(children, (child) => React.cloneElement(child, { isDragDisabled }));
+
+  return (
+    <DragDropContext onDragEnd={onDragEnd}>
+      <Droppable droppableId='droppable'>
+        {(provided) => (
```
Comment (maximsan) [inline src/components/DraggableTable/DraggableTable.js:24]: spread it ?

File: src/components/DraggableTable/DraggableTable.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+import { bindActionCreators } from 'redux';
+import { DragDropContext, Droppable } from 'react-beautiful-dnd';
+import { reorderTable } from '../../store/actions';
+
+const DraggableTable = ({ tableData, reorderTable, children, tableType, userId }) => {
+  const onDragEnd = ({ destination, source }) => {
+    if (!destination) return;
+
+    reorderTable(tableType, tableData, source.index, destination.index, userId);
+  };
```
Comment (maximsan) [inline src/components/DraggableTable/DraggableTable.js:15]: I guess it worth to wrap in useCallback

File: src/store/actions/dataActions.js [hunk 5]
```diff
@@ -327,6 +331,17 @@ const sendMail = (mailData) => {
   };
 };
 
+const reorderTable = (table, list, startIndex, endIndex, userId) => {
+  const result = Array.from(list);
+  const [removed] = result.splice(startIndex, 1);
```
Comment (maximsan) [inline src/store/actions/dataActions.js:336]: splice mutates data, do you need it ? try to use slice or rest instead

File: src/store/actions/dataActions.js [hunk 6]
```diff
@@ -327,6 +331,17 @@ const sendMail = (mailData) => {
...
+
+  result.splice(endIndex, 0, removed);
```
Comment (maximsan) [inline src/store/actions/dataActions.js:338]: and use here use concat or rest

File: src/utils/dragAndDropHelpers.js [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+  if (order) {
+    const idName = dataIdNames[keyWithoutId];
+
+    const sortedData = [...data];
+    sortedData.sort((a, b) => order.indexOf(a[idName]) - order.indexOf(b[idName]));
+    return sortedData;
+  }
+
+  return data;
+};
```
Comment (maximsan) [inline src/utils/dragAndDropHelpers.js:34]: add unit-tests

File: src/components/DraggableRow/DraggableRow.js (hunk 2 above)
Comment (MaxDaniLOVE) [reply src/components/DraggableRow/DraggableRow.js:8]: done

File: src/components/DraggableTable/DraggableTable.js (hunk 3 above)
Comment (MaxDaniLOVE) [reply src/components/DraggableTable/DraggableTable.js:24]: done

File: src/components/DraggableTable/DraggableTable.js (hunk 4 above)
Comment (MaxDaniLOVE) [reply src/components/DraggableTable/DraggableTable.js:15]: done

File: src/store/actions/dataActions.js (hunk 5 above)
Comment (MaxDaniLOVE) [reply src/store/actions/dataActions.js:336]: done

File: src/store/actions/dataActions.js (hunk 6 above)
Comment (MaxDaniLOVE) [reply src/store/actions/dataActions.js:338]: done

**Pull request merged**